src/ceph-disk

```diff
@@ -1166,6 +1698,48 @@ def extract_parted_partition_numbers(partitions):
     numbers_as_strings = re.findall('^\d+', partitions, re.MULTILINE)
     return map(int, numbers_as_strings)
 
+
+# wait for udev event queue to clear
+def udev_settle():
+    time.sleep(1)
```

Comment:
What is the rationale for this sleep ? If something fails when this is removed, it means there is a race condition somewhere that needs to be addressed. If a problem is resolved on a given platform by adding a sleep, it means it will fail on a platform that is slower or faster than the one you're using.